mergify_engine/context.py

```diff
@@ -220,27 +230,29 @@ async def clear_team_members_cache_for_org(
     async def get_team_members(
         self, team_slug: github_types.GitHubTeamSlug
     ) -> typing.List[github_types.GitHubLogin]:
-
-        key = self._team_members_cache_key_for_repo(self.owner_id)
-        members_raw = await self.redis.hget(key, team_slug)
-        if members_raw is None:
-            members = [
-                github_types.GitHubLogin(member["login"])
-                async for member in self.client.items(
-                    f"/orgs/{self.owner_login}/teams/{team_slug}/members"
+        members = self._cache.team_members.get(team_slug, NotCached)
```

Comment:
Not a fan of having `NotCached` pass as an arg to `get`. It'd be better that `get` returns `NotCached` if it's not cached.

e.g. https://github.com/Mergifyio/mergify-dashboard/blob/main/mergify_dashboard/cache.py

Reply:
I think you mix dashboard redis cache and engine in-memory cache. `self._cache.team_members` is a simple dict here. We pass `NotCached` to `.get()` to differentiate cached `None` and not cached.

Reply:
I don't mix anything, I'm just showing an example of a `Cache` class.

You shouldn't use a raw `dict` here because you are leaking the cache abstraction in the user by having it pass `NotCached` as the default argument.

Something like this would be cleaner:
```python
class Cache(dict[K, V]):
    def get(self, key: K) -> typing.Union[V, NotCached]:
        ...
```
(You can add support for the `default` arg but not sure you'd need it).

Reply:
Actually `NotCached` could be anything it's just a marker that you are abusing here.

So alternatively, if you want, you should rewrite the code like this:

```python
try:
   members = self._cache.team_members[team_slug]
except KeyError
   key = self._team_members_cache_key_for_repo(self.owner_id)
   ...
```

In that case, you don't need to use a marker, `NotCached` or any other.